File: backend/app/api/routes/transits.py
"""
Transit API Routes
Provides endpoints for transit calculations and analysis
"""
from datetime import datetime, timedelta
from typing import Optional, List
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from pydantic import BaseModel, Field

from app.core.database_sqlite import get_db
from app.models.birth_data import BirthData
from app.models.chart import Chart
from app.services.transit_calculator import TransitCalculator, TransitInterpreter
from app.services.chart_calculator import NatalChartCalculator

router = APIRouter()


# Request/Response schemas
class TransitRequest(BaseModel):
    birth_data_id: str
    transit_date: Optional[str] = None  # ISO format, defaults to now
    zodiac: str = "tropical"
    orb_multiplier: float = Field(default=1.0, ge=0.5, le=2.0)


class TransitTimelineRequest(BaseModel):
    birth_data_id: str
    start_date: str  # ISO format
    end_date: str    # ISO format
    zodiac: str = "tropical"
    interval_days: int = Field(default=1, ge=1, le=7)


class ExactTransitRequest(BaseModel):
    birth_data_id: str
    transit_planet: str
    natal_planet: str
    aspect: str
    start_date: str
    end_date: str
    zodiac: str = "tropical"


class TransitAspect(BaseModel):
    transit_planet: str
    natal_planet: str
    aspect: str
    orb: float
    is_applying: bool
    significance: str
    estimated_duration: str
    transit_sign: str
    transit_degree: float
    natal_sign: str
    natal_degree: float
    transit_retrograde: bool


class TransitSummary(BaseModel):
    total_transits: int
    major_count: int
    significant_count: int
    aspect_counts: dict
    themes: List[str]
    most_significant: Optional[TransitAspect] = None


class TransitResponse(BaseModel):
    transit_datetime: str
    current_positions: dict
    transits: List[TransitAspect]
    summary: TransitSummary


def get_natal_chart_data(birth_data_id: str, db: Session, zodiac: str = "tropical") -> dict:
    """Helper to get natal chart data for transit calculations."""
    birth_data = db.query(BirthData).filter(BirthData.id == birth_data_id).first()
    if not birth_data:
        raise HTTPException(status_code=404, detail="Birth data not found")

    # Parse birth datetime
    birth_dt = datetime.strptime(
        f"{birth_data.birth_date} {birth_data.birth_time or '12:00:00'}",
        "%Y-%m-%d %H:%M:%S"
    )

    # Calculate natal chart
    chart_data = NatalChartCalculator.calculate_natal_chart(
        birth_datetime=birth_dt,
        latitude=birth_data.latitude,
        longitude=birth_data.longitude,
        timezone_offset_minutes=birth_data.utc_offset or 0,
        zodiac=zodiac
    )

    return chart_data


@router.post("/current", response_model=TransitResponse)
async def get_current_transits(
    request: TransitRequest,
    db: Session = Depends(get_db)
):
    """
    Get current transits for a birth chart.

    Returns all transiting planets' aspects to natal positions,
    sorted by significance.
    """
    # Get natal chart
    chart_data = get_natal_chart_data(request.birth_data_id, db, request.zodiac)

    # Parse transit date
    transit_dt = None
    if request.transit_date:
        transit_dt = datetime.fromisoformat(request.transit_date.replace('Z', '+00:00'))

    # Calculate transits
    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        transit_datetime=transit_dt,
        zodiac=request.zodiac,
        orb_multiplier=request.orb_multiplier
    )

    return transits


@router.get("/current/{birth_data_id}")
async def get_current_transits_simple(
    birth_data_id: str,
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """Simple GET endpoint for current transits."""
    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        zodiac=zodiac
    )

    return transits


@router.post("/timeline")
async def get_transit_timeline(
    request: TransitTimelineRequest,
    db: Session = Depends(get_db)
):
    """
    Get transit timeline over a date range.

    Returns significant transits for each day in the range,
    useful for visualization and planning.
    """
    chart_data = get_natal_chart_data(request.birth_data_id, db, request.zodiac)

    start = datetime.fromisoformat(request.start_date.replace('Z', '+00:00'))
    end = datetime.fromisoformat(request.end_date.replace('Z', '+00:00'))

    # Limit range to prevent abuse
    if (end - start).days > 365:
        raise HTTPException(
            status_code=400,
            detail="Date range cannot exceed 365 days"
        )

    timeline = TransitCalculator.calculate_transit_timeline(
        natal_planets=chart_data['planets'],
        start_date=start,
        end_date=end,
        zodiac=request.zodiac,
        interval_days=request.interval_days
    )

    return {
        "start_date": request.start_date,
        "end_date": request.end_date,
        "timeline": timeline,
        "total_days": len(timeline)
    }


@router.get("/upcoming/{birth_data_id}")
async def get_upcoming_transits(
    birth_data_id: str,
    days: int = Query(default=30, ge=7, le=90),
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """
    Get upcoming significant transits.

    Returns major and significant transits approaching
    over the next N days.
    """
    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    upcoming = TransitCalculator.get_upcoming_significant_transits(
        natal_planets=chart_data['planets'],
        days_ahead=days,
        zodiac=zodiac
    )

    return {
        "days_ahead": days,
        "upcoming_transits": upcoming,
        "count": len(upcoming)
    }


@router.post("/exact-dates")
async def find_exact_transit_dates(
    request: ExactTransitRequest,
    db: Session = Depends(get_db)
):
    """
    Find exact dates for a specific transit aspect.

    Useful for planning around significant transits
    like Saturn conjunct Sun, etc.
    """
    chart_data = get_natal_chart_data(request.birth_data_id, db, request.zodiac)

    start = datetime.fromisoformat(request.start_date.replace('Z', '+00:00'))
    end = datetime.fromisoformat(request.end_date.replace('Z', '+00:00'))

    # Limit range
    if (end - start).days > 730:
        raise HTTPException(
            status_code=400,
            detail="Date range cannot exceed 2 years"
        )

    exact_dates = TransitCalculator.find_exact_transit_dates(
        natal_planets=chart_data['planets'],
        transit_planet=request.transit_planet,
        natal_planet=request.natal_planet,
        aspect=request.aspect,
        start_date=start,
        end_date=end,
        zodiac=request.zodiac
    )

    return {
        "transit_planet": request.transit_planet,
        "natal_planet": request.natal_planet,
        "aspect": request.aspect,
        "exact_dates": exact_dates,
        "count": len(exact_dates)
    }


@router.get("/interpret/{transit_planet}/{natal_planet}/{aspect}")
async def get_transit_interpretation(
    transit_planet: str,
    natal_planet: str,
    aspect: str
):
    """
    Get interpretation for a specific transit aspect.

    Returns theme, description, duration, and advice.
    """
    interpretation = TransitInterpreter.get_interpretation(
        transit_planet, natal_planet, aspect
    )

    return interpretation


@router.get("/report/{birth_data_id}")
async def get_transit_report(
    birth_data_id: str,
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """
    Get a text report of current transits.

    Provides a summary suitable for display or AI interpretation.
    """
    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        zodiac=zodiac
    )

    report = TransitInterpreter.get_transit_report(transits['transits'])

    return {
        "transit_datetime": transits['transit_datetime'],
        "report": report,
        "summary": transits['summary']
    }


@router.get("/daily-snapshot/{birth_data_id}")
async def get_daily_transit_snapshot(
    birth_data_id: str,
    date: Optional[str] = Query(default=None),
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """
    Get a snapshot of transits for a specific day.

    Includes current positions, active transits, and themes.
    Useful for daily transit updates.
    """
    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    transit_dt = None
    if date:
        transit_dt = datetime.fromisoformat(date.replace('Z', '+00:00'))
    else:
        transit_dt = datetime.utcnow().replace(hour=12, minute=0, second=0, microsecond=0)

    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        transit_datetime=transit_dt,
        zodiac=zodiac
    )

    # Get Moon phase (simplified)
    moon_lon = transits['current_positions'].get('Moon', {}).get('longitude', 0)
    sun_lon = transits['current_positions'].get('Sun', {}).get('longitude', 0)
    moon_phase_angle = (moon_lon - sun_lon) % 360

    moon_phase = "New Moon"
    if 45 < moon_phase_angle <= 90:
        moon_phase = "Waxing Crescent"
    elif 90 < moon_phase_angle <= 135:
        moon_phase = "First Quarter"
    elif 135 < moon_phase_angle <= 180:
        moon_phase = "Waxing Gibbous"
    elif 180 < moon_phase_angle <= 225:
        moon_phase = "Full Moon"
    elif 225 < moon_phase_angle <= 270:
        moon_phase = "Waning Gibbous"
    elif 270 < moon_phase_angle <= 315:
        moon_phase = "Last Quarter"
    elif 315 < moon_phase_angle <= 360:
        moon_phase = "Waning Crescent"

    return {
        "date": transit_dt.date().isoformat(),
        "moon_phase": moon_phase,
        "moon_sign": transits['current_positions'].get('Moon', {}).get('sign_name', ''),
        "sun_sign": transits['current_positions'].get('Sun', {}).get('sign_name', ''),
        "active_transits": transits['transits'][:10],  # Top 10
        "themes": transits['summary']['themes'],
        "major_transit": transits['summary']['most_significant']
    }


# ==================== AI Transit Interpretation Endpoints ====================

@router.post("/ai/interpret-transit")
async def ai_interpret_transit(
    transit_data: dict,
    db: Session = Depends(get_db)
):
    """
    Get AI-generated interpretation for a specific transit.

    Requires ANTHROPIC_API_KEY to be set.
    """
    from app.services.ai_interpreter import AIInterpreter

    try:
        interpreter = AIInterpreter()
        interpretation = await interpreter.generate_transit_interpretation_async(transit_data)

        return {
            "transit": f"{transit_data.get('transit_planet')} {transit_data.get('aspect')} {transit_data.get('natal_planet')}",
            "interpretation": interpretation
        }
    except ValueError as e:
        raise HTTPException(status_code=503, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"AI interpretation failed: {str(e)}")


@router.get("/ai/daily-forecast/{birth_data_id}")
async def ai_daily_forecast(
    birth_data_id: str,
    date: Optional[str] = Query(default=None),
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """
    Get AI-generated daily transit forecast.

    Combines the daily snapshot with AI interpretation
    for a personalized daily horoscope.
    """
    from app.services.ai_interpreter import AIInterpreter

    # Get daily snapshot
    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    transit_dt = None
    if date:
        transit_dt = datetime.fromisoformat(date.replace('Z', '+00:00'))
    else:
        transit_dt = datetime.utcnow().replace(hour=12, minute=0, second=0, microsecond=0)

    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        transit_datetime=transit_dt,
        zodiac=zodiac
    )

    # Calculate moon phase
    moon_lon = transits['current_positions'].get('Moon', {}).get('longitude', 0)
    sun_lon = transits['current_positions'].get('Sun', {}).get('longitude', 0)
    moon_phase_angle = (moon_lon - sun_lon) % 360

    moon_phase = "New Moon"
    if 45 < moon_phase_angle <= 90:
        moon_phase = "Waxing Crescent"
    elif 90 < moon_phase_angle <= 135:
        moon_phase = "First Quarter"
    elif 135 < moon_phase_angle <= 180:
        moon_phase = "Waxing Gibbous"
    elif 180 < moon_phase_angle <= 225:
        moon_phase = "Full Moon"
    elif 225 < moon_phase_angle <= 270:
        moon_phase = "Waning Gibbous"
    elif 270 < moon_phase_angle <= 315:
        moon_phase = "Last Quarter"
    elif 315 < moon_phase_angle <= 360:
        moon_phase = "Waning Crescent"

    daily_snapshot = {
        "date": transit_dt.date().isoformat(),
        "moon_phase": moon_phase,
        "moon_sign": transits['current_positions'].get('Moon', {}).get('sign_name', ''),
        "sun_sign": transits['current_positions'].get('Sun', {}).get('sign_name', ''),
        "active_transits": transits['transits'][:10],
        "themes": transits['summary']['themes'],
        "major_transit": transits['summary']['most_significant']
    }

    try:
        interpreter = AIInterpreter()
        forecast = await interpreter.generate_daily_transit_forecast_async(daily_snapshot)

        return {
            "date": daily_snapshot["date"],
            "moon_phase": moon_phase,
            "moon_sign": daily_snapshot["moon_sign"],
            "sun_sign": daily_snapshot["sun_sign"],
            "themes": daily_snapshot["themes"],
            "major_transit": daily_snapshot["major_transit"],
            "forecast": forecast,
            "active_transit_count": len(transits['transits'])
        }
    except ValueError as e:
        raise HTTPException(status_code=503, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"AI forecast generation failed: {str(e)}")


@router.get("/ai/transit-report/{birth_data_id}")
async def ai_transit_report(
    birth_data_id: str,
    report_type: str = Query(default="comprehensive", pattern="^(comprehensive|highlights|brief)$"),
    zodiac: str = Query(default="tropical"),
    db: Session = Depends(get_db)
):
    """
    Get AI-generated comprehensive transit report.

    Report types:
    - comprehensive: Full detailed analysis
    - highlights: Key transits focus
    - brief: Quick summary
    """
    from app.services.ai_interpreter import AIInterpreter

    chart_data = get_natal_chart_data(birth_data_id, db, zodiac)

    transits = TransitCalculator.calculate_current_transits(
        natal_planets=chart_data['planets'],
        zodiac=zodiac
    )

    try:
        interpreter = AIInterpreter()
        report = await interpreter.generate_transit_report_async(transits, report_type=report_type)

        return {
            "transit_datetime": transits['transit_datetime'],
            "report_type": report_type,
            "summary": transits['summary'],
            "report": report
        }
    except ValueError as e:
        raise HTTPException(status_code=503, detail=str(e))
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"AI report generation failed: {str(e)}")

File: frontend/src/lib/api/transits.ts
/**
 * Transit API client
 *
 * Part of Phase 3: Advanced Transit Analysis
 */
import { apiClient, getErrorMessage } from './client'

// Types
export interface TransitAspect {
  transit_planet: string
  natal_planet: string
  aspect: string
  orb: number
  is_applying: boolean
  significance: 'major' | 'significant' | 'moderate' | 'minor'
  estimated_duration: string
  transit_sign: string
  transit_degree: number
  natal_sign: string
  natal_degree: number
  transit_retrograde: boolean
}

export interface TransitSummary {
  total_transits: number
  major_count: number
  significant_count: number
  aspect_counts: Record<string, number>
  themes: string[]
  most_significant: TransitAspect | null
}

export interface PlanetPosition {
  longitude: number
  latitude: number
  distance: number
  speed_longitude: number
  is_retrograde: boolean
  sign_name: string
  degree_in_sign: number
}

export interface CurrentTransitsResponse {
  transit_datetime: string
  current_positions: Record<string, PlanetPosition>
  transits: TransitAspect[]
  summary: TransitSummary
}

export interface TransitTimelineEntry {
  date: string
  transits: TransitAspect[]
}

export interface TransitTimelineResponse {
  start_date: string
  end_date: string
  timeline: TransitTimelineEntry[]
  total_days: number
}

export interface UpcomingTransit extends TransitAspect {
  first_date: string
}

export interface UpcomingTransitsResponse {
  days_ahead: number
  upcoming_transits: UpcomingTransit[]
  count: number
}

export interface ExactTransitDate {
  date: string
  transit_planet: string
  natal_planet: string
  aspect: string
  retrograde: boolean
}

export interface ExactTransitDatesResponse {
  transit_planet: string
  natal_planet: string
  aspect: string
  exact_dates: ExactTransitDate[]
  count: number
}

export interface TransitInterpretation {
  theme: string
  description: string
  duration: string
  advice: string
}

export interface TransitReportResponse {
  transit_datetime: string
  report: string
  summary: TransitSummary
}

export interface DailySnapshotResponse {
  date: string
  moon_phase: string
  moon_sign: string
  sun_sign: string
  active_transits: TransitAspect[]
  themes: string[]
  major_transit: TransitAspect | null
}

// Request types
export interface TransitRequest {
  birth_data_id: string
  transit_date?: string
  zodiac?: 'tropical' | 'sidereal'
  orb_multiplier?: number
}

export interface TransitTimelineRequest {
  birth_data_id: string
  start_date: string
  end_date: string
  zodiac?: 'tropical' | 'sidereal'
  interval_days?: number
}

export interface ExactTransitRequest {
  birth_data_id: string
  transit_planet: string
  natal_planet: string
  aspect: string
  start_date: string
  end_date: string
  zodiac?: 'tropical' | 'sidereal'
}

// API Functions

/**
 * Get current transits for a birth chart
 */
export async function getCurrentTransits(
  request: TransitRequest
): Promise<CurrentTransitsResponse> {
  try {
    const response = await apiClient.post('/transits/current', request)
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Simple GET for current transits
 */
export async function getCurrentTransitsSimple(
  birthDataId: string,
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<CurrentTransitsResponse> {
  try {
    const response = await apiClient.get(`/transits/current/${birthDataId}`, {
      params: { zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get transit timeline over a date range
 */
export async function getTransitTimeline(
  request: TransitTimelineRequest
): Promise<TransitTimelineResponse> {
  try {
    const response = await apiClient.post('/transits/timeline', request)
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get upcoming significant transits
 */
export async function getUpcomingTransits(
  birthDataId: string,
  days: number = 30,
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<UpcomingTransitsResponse> {
  try {
    const response = await apiClient.get(`/transits/upcoming/${birthDataId}`, {
      params: { days, zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Find exact dates for a specific transit
 */
export async function findExactTransitDates(
  request: ExactTransitRequest
): Promise<ExactTransitDatesResponse> {
  try {
    const response = await apiClient.post('/transits/exact-dates', request)
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get interpretation for a transit aspect
 */
export async function getTransitInterpretation(
  transitPlanet: string,
  natalPlanet: string,
  aspect: string
): Promise<TransitInterpretation> {
  try {
    const response = await apiClient.get(
      `/transits/interpret/${transitPlanet}/${natalPlanet}/${aspect}`
    )
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get text report of current transits
 */
export async function getTransitReport(
  birthDataId: string,
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<TransitReportResponse> {
  try {
    const response = await apiClient.get(`/transits/report/${birthDataId}`, {
      params: { zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get daily transit snapshot
 */
export async function getDailySnapshot(
  birthDataId: string,
  date?: string,
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<DailySnapshotResponse> {
  try {
    const response = await apiClient.get(`/transits/daily-snapshot/${birthDataId}`, {
      params: { date, zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

// Helper functions

/**
 * Get color for transit significance
 */
export function getSignificanceColor(significance: string): string {
  const colors: Record<string, string> = {
    major: 'text-celestial-pink',
    significant: 'text-celestial-gold',
    moderate: 'text-blue-400',
    minor: 'text-gray-400'
  }
  return colors[significance] || 'text-gray-400'
}

/**
 * Get symbol for aspect
 */
export function getAspectSymbol(aspect: string): string {
  const symbols: Record<string, string> = {
    conjunction: '☌',
    opposition: '☍',
    trine: '△',
    square: '□',
    sextile: '⚹',
    quincunx: '⚻'
  }
  return symbols[aspect] || aspect
}

/**
 * Get astrological glyph for planet
 */
export function getPlanetSymbol(planet: string): string {
  // Normalize planet name (lowercase, handle variations)
  const normalized = planet.toLowerCase().replace(/_/g, ' ').trim()

  const symbols: Record<string, string> = {
    'sun': '☉',
    'moon': '☽',
    'mercury': '☿',
    'venus': '♀',
    'mars': '♂',
    'jupiter': '♃',
    'saturn': '♄',
    'uranus': '♅',
    'neptune': '♆',
    'pluto': '♇',
    'chiron': '⚷',
    'north node': '☊',
    'south node': '☋',
    'ascendant': 'Asc',
    'midheaven': 'MC',
    'lilith': '⚸',
  }

  return symbols[normalized] || planet
}

/**
 * Get planet display name (capitalized)
 */
export function getPlanetDisplayName(planet: string): string {
  // Handle snake_case like "north_node"
  return planet
    .split('_')
    .map(word => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
    .join(' ')
}

/**
 * Format transit for display (with symbols)
 */
export function formatTransit(transit: TransitAspect): string {
  const transitSymbol = getPlanetSymbol(transit.transit_planet)
  const aspectSymbol = getAspectSymbol(transit.aspect)
  const natalSymbol = getPlanetSymbol(transit.natal_planet)
  const status = transit.is_applying ? '→' : '←'
  const retro = transit.transit_retrograde ? '℞' : ''
  return `${transitSymbol}${retro} ${aspectSymbol} ${natalSymbol} (${transit.orb.toFixed(1)}° ${status})`
}

/**
 * Format transit with full planet names (for accessibility/tooltips)
 */
export function formatTransitFull(transit: TransitAspect): string {
  const status = transit.is_applying ? 'applying' : 'separating'
  const retro = transit.transit_retrograde ? ' (retrograde)' : ''
  return `${getPlanetDisplayName(transit.transit_planet)}${retro} ${transit.aspect} ${getPlanetDisplayName(transit.natal_planet)} - ${transit.orb.toFixed(1)}° ${status}`
}

/**
 * Get theme description
 */
export function getThemeDescription(theme: string): string {
  const descriptions: Record<string, string> = {
    transformation: 'Deep changes and rebirth',
    expansion: 'Growth and opportunities',
    awakening: 'Spiritual insights and breakthroughs',
    challenge: 'Tests and obstacles to overcome',
    opportunity: 'Favorable conditions for action'
  }
  return descriptions[theme] || theme
}

// ==================== AI Transit Functions ====================

export interface AITransitInterpretationResponse {
  transit: string
  interpretation: string
}

export interface AIDailyForecastResponse {
  date: string
  moon_phase: string
  moon_sign: string
  sun_sign: string
  themes: string[]
  major_transit: TransitAspect | null
  forecast: string
  active_transit_count: number
}

export interface AITransitReportResponse {
  transit_datetime: string
  report_type: 'comprehensive' | 'highlights' | 'brief'
  summary: TransitSummary
  report: string
}

/**
 * Get AI interpretation for a specific transit
 */
export async function getAITransitInterpretation(
  transitData: TransitAspect
): Promise<AITransitInterpretationResponse> {
  try {
    const response = await apiClient.post('/transits/ai/interpret-transit', transitData)
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get AI-generated daily transit forecast
 */
export async function getAIDailyForecast(
  birthDataId: string,
  date?: string,
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<AIDailyForecastResponse> {
  try {
    const response = await apiClient.get(`/transits/ai/daily-forecast/${birthDataId}`, {
      params: { date, zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}

/**
 * Get AI-generated comprehensive transit report
 */
export async function getAITransitReport(
  birthDataId: string,
  reportType: 'comprehensive' | 'highlights' | 'brief' = 'comprehensive',
  zodiac: 'tropical' | 'sidereal' = 'tropical'
): Promise<AITransitReportResponse> {
  try {
    const response = await apiClient.get(`/transits/ai/transit-report/${birthDataId}`, {
      params: { report_type: reportType, zodiac }
    })
    return response.data
  } catch (error) {
    throw new Error(getErrorMessage(error))
  }
}
